Moved bump methods from Dragger to Controller.

AnmAtAnm opened the pull request:
Moved bump methods from Dragger to Controller. Also moving more logic into .connect(..) call, in prep for events. It now takes one less parameter and includes call to bumpNeighbors(), which now include requestLayout().

Resolves #82.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/google/blockly-android/337)
<!-- Reviewable:end -->

RoboErikG commented:
Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/control/BlocklyController.java, line 1117 \[r3\]](https://reviewable.io:443/reviews/google/blockly-android/337#-KM0OmoGcTt5rH1BBLpZ:-KM0OmoGcTt5rH1BBLp_:-1080162791) ([raw file](https://github.com/google/blockly-android/blob/872bcce0c7f5fe04916520488b2d1b8b45c94f42/blocklylib-core/src/main/java/com/google/blockly/android/control/BlocklyController.java#L1117)):*
> ```Java
> 
>     /**
>      * This implements {@link #connect(Connection, Connection)}.  It is not responsible for firing
> ```

It calls firePendingEvents at the end. Could you clarify the firing events comment?

---

*[blocklylib-core/src/main/java/com/google/blockly/android/control/BlocklyController.java, line 1168 \[r3\]](https://reviewable.io:443/reviews/google/blockly-android/337#-KM0Ou6WPkfnE6iecLki:-KM0Ou6WPkfnE6iecLkj:-1650662300) ([raw file](https://github.com/google/blockly-android/blob/872bcce0c7f5fe04916520488b2d1b8b45c94f42/blocklylib-core/src/main/java/com/google/blockly/android/control/BlocklyController.java#L1168)):*
> ```Java
>         } // otherwise we are probably running headless, without views.
> 
>         firePendingEvents();
> ```

Did you mean to have this here or are the callers responsible?

---

*[blocklylib-core/src/main/java/com/google/blockly/android/ui/Dragger.java, line 349 \[r3\]](https://reviewable.io:443/reviews/google/blockly-android/337#-KM0Pddz-X7p0K5QLupA:-KM0Pddz-X7p0K5QLupB:1462306306) ([raw file](https://github.com/google/blockly-android/blob/872bcce0c7f5fe04916520488b2d1b8b45c94f42/blocklylib-core/src/main/java/com/google/blockly/android/ui/Dragger.java#L349)):*
> ```Java
>             mConnectionManager.addConnection(cur);
>         }
>         mDraggedConnections.clear();
> ```

I think you still need to get the new root BlockGroup and call requestLayout so all the connector positions are updated after this finishes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/337)*
<!-- Sent from Reviewable.io -->

AnmAtAnm commented:
PTAL @roboerikg 

---

Review status: 0 of 5 files reviewed at latest revision, 3 unresolved discussions.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/control/BlocklyController.java, line 1117 \[r3\]](https://reviewable.io:443/reviews/google/blockly-android/337#-KM0OmoGcTt5rH1BBLpZ:-KM0RAYvLrdxBpXaSBne:247530640) ([raw file](https://github.com/google/blockly-android/blob/872bcce0c7f5fe04916520488b2d1b8b45c94f42/blocklylib-core/src/main/java/com/google/blockly/android/control/BlocklyController.java#L1117)):*
<details><summary><i>Previously, RoboErikG wrote…</i></summary>
> It calls firePendingEvents at the end. Could you clarify the firing events comment?
</details>
Clarified comment. Removed firePendingEvents().

---

*[blocklylib-core/src/main/java/com/google/blockly/android/control/BlocklyController.java, line 1168 \[r3\]](https://reviewable.io:443/reviews/google/blockly-android/337#-KM0Ou6WPkfnE6iecLki:-KM0RHE02-8DEylWovS4:1740281843) ([raw file](https://github.com/google/blockly-android/blob/872bcce0c7f5fe04916520488b2d1b8b45c94f42/blocklylib-core/src/main/java/com/google/blockly/android/control/BlocklyController.java#L1168)):*
<details><summary><i>Previously, RoboErikG wrote…</i></summary>
> Did you mean to have this here or are the callers responsible?
</details>
Removed.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/ui/Dragger.java, line 349 \[r3\]](https://reviewable.io:443/reviews/google/blockly-android/337#-KM0Pddz-X7p0K5QLupA:-KM0RN25Gk0_m54K92m_:-2030689613) ([raw file](https://github.com/google/blockly-android/blob/872bcce0c7f5fe04916520488b2d1b8b45c94f42/blocklylib-core/src/main/java/com/google/blockly/android/ui/Dragger.java#L349)):*
<details><summary><i>Previously, RoboErikG wrote…</i></summary>
> I think you still need to get the new root BlockGroup and call requestLayout so all the connector positions are updated after this finishes.
</details>
These calls still exist.  They just moved into the bumpNeighbors(..) method, which is now called by the connect(..) method.  The goal is to have an event group capture by a single controller call. Here, the bump being a side effect of the connect.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/337#-:-KLy9B8l5Leei_0_-4pK:1290518790)*
<!-- Sent from Reviewable.io -->

RoboErikG commented:
One issue, otherwise <img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

---

Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/control/BlocklyController.java, line 802 \[r4\]](https://reviewable.io:443/reviews/google/blockly-android/337#-KM28XsqCbSl-hC4g5sT:-KM28XsrO2F609KFoodf:-1559616531) ([raw file](https://github.com/google/blockly-android/blob/bf27201840131fe7268aafad6ee8fbe6c0b78303/blocklylib-core/src/main/java/com/google/blockly/android/control/BlocklyController.java#L802)):*
> ```Java
>      * @return True if a zoom was changed. Otherwise false.
>      */
>     public boolean zoomIn() {
> ```

I'm not sure zoom belongs in this class. It's purely a UI function.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/337#-:-KM2DCQxlY2r3gvIJs8-:1026886312)*
<!-- Sent from Reviewable.io -->
